src/io/flutter/logging/FlutterLogView.java

```diff
@@ -45,20 +78,26 @@
     public SimpleTextAttributes style(@Nullable FlutterLogEntry entry, int attributes) {
       if (showColors && entry != null) {
         final FlutterLog.Level level = FlutterLog.Level.forValue(entry.getLevel());
-        // TODO(quangson91): use color from FlutterLogColorPage
         if (level != null) {
-          switch (level) {
-            case FINER:
-              return SimpleTextAttributes.GRAY_ATTRIBUTES;
-            case WARNING:
-              return ORANGE_ATTRIBUTES;
-            case SEVERE:
-              return SimpleTextAttributes.ERROR_ATTRIBUTES;
-          }
+          return getTextAttributesByLogLevel(level);
         }
       }
       return SimpleTextAttributes.REGULAR_ATTRIBUTES;
     }
+
+    @NotNull
+    private SimpleTextAttributes getTextAttributesByLogLevel(@NotNull Level level) {
+      try {
+        final TextAttributesKey key = LOG_LEVEL_TEXT_ATTRIBUTES_KEY_MAP.get(level);
+        final Color color = globalEditorColorsScheme.getAttributes(key).getForegroundColor();
+        return new SimpleTextAttributes(SimpleTextAttributes.STYLE_PLAIN, color);
```

Comment:
Does this create a new `SimpleTextAttributes` object for each entry?  If so, would it make sense to cache them?

Reply:
Ah, you right.
We should be caching it. So we don't have re-created new `SimpleTextAttributes` value.
I implemented in this commit: https://github.com/flutter/flutter-intellij/pull/2399/commits/72ab8ff2668a61c0ff07747e2bca5dde782078ba

Please take a look!